### PR TITLE
feat: add getGasPrice rpc provider method

### LINKED ISSUE #761: To have a method : provider.getGasprice()
**Is your feature request related to a problem? Please describe.**
Improvement :
Rpc specification v0.5.0 will add : 
```
 "l1_gas_price": {
                        "title": "L1 gas price",
                        "descritpion": "The price of l1 gas in the block",
                        "$ref": "#/components/schemas/RESOURCE_PRICE"
                    },
```
It could be helpful to DAPP devs to have an easy access to this value.

**Describe the solution you'd like**
Something like this : 
```typescript
const GasPrice = provider.getGasPrice();
```

**Describe alternatives you've considered**
"alternative facts" ?

**Additional context**
To consider when PR for rpc 0.5.0 will be merged.
I can easily propose a PR on this subject. 
Probably a subject that will have to be handled and to be customizable in starknet-devnet-rs.

### PR DESCRIPTION
**Pull request by princeibs**
## Motivation and Resolution
Closes #761 

### RPC version (if applicable)

...

## Usage related changes

...

## Development related changes

...

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

**Comment by tabaktoni** [comment]
@PhilippeR26

File: src/provider/interface.ts [hunk 1]
```diff
@@ -78,6 +78,14 @@ export abstract class ProviderInterface {
     blockIdentifier?: BlockIdentifier
   ): Promise<ContractClassResponse>;
 
+  /**
+   * Gets the price of l1 gas in the block
+   *
+   * @param blockIdentifier block identifier
+   * @returns gas price of the block
+   */
+  public abstract getGasPrice(blockIdentifier: BlockIdentifier): Promise<string>;
```
Comment (ivpavici) [inline src/provider/interface.ts:87]: do we want to be completely explicit and rename to `getL1GasPrice` ?

File: __tests__/rpcProvider.test.ts [hunk 2]
```diff
@@ -71,6 +71,11 @@ describeIfRpc('RPCProvider', () => {
     expect(typeof blockNumber).toBe('number');
   });
 
+  test('getGasprice', async () => {
+    const gasPrice = await rpcProvider.getGasPrice(967555);
```
Comment (ivpavici) [inline __tests__/rpcProvider.test.ts:75]: This test cannot work on devnet... 

24: Block not found: undefined

Comment (ivpavici) [inline __tests__/rpcProvider.test.ts:75]: This test cannot work on devnet...

24: Block not found: undefined

**Review by ivpavici — CHANGES_REQUESTED** [review]

File: __tests__/rpcProvider.test.ts (hunk 2 above)
Comment (PhilippeR26) [reply __tests__/rpcProvider.test.ts:75]: "latest" as parameter should be OK.

File: src/provider/rpc.ts [hunk 3]
```diff
@@ -100,6 +100,12 @@ export class RpcProvider implements ProviderInterface {
     return this.channel.getBlockWithTxs(blockIdentifier);
   }
 
+  public async getGasPrice(blockIdentifier?: BlockIdentifier) {
+    return this.channel
+      .getBlockWithTxs(blockIdentifier)
```
Comment (PhilippeR26) [inline src/provider/rpc.ts:105]: `getBlockWithTxHashes` should generate less amount of data to receive, and will nevertheless return the L1 gas price

File: src/utils/responseParser/rpc.ts [hunk 4]
```diff
@@ -117,4 +118,8 @@ export class RPCResponseParser
       abi: isString(res.abi) ? JSON.parse(res.abi) : res.abi,
     };
   }
+
+  public parseGasPriceResponse(res: BlockWithTxs): string {
```
Comment (PhilippeR26) [inline src/utils/responseParser/rpc.ts:122]: here also, add the L1 word will clarify the usage : parseL1GasPriceResponse

**Review by PhilippeR26 — CHANGES_REQUESTED** [review]
Just few improvements and variables naming.

File: src/provider/rpc.ts (hunk 3 above)
Comment (princeibs) [reply src/provider/rpc.ts:105]: I checked the response from BlastAPI but I couldn't find details of l1 gas price: 
image

**Comment by princeibs** [comment]
@ivpavici @PhilippeR26

**Review by ivpavici — APPROVED** [review]
LGTM
let's see about @PhilippeR26 comment still

File: src/provider/rpc.ts (hunk 3 above)
Comment (PhilippeR26) [reply src/provider/rpc.ts:105]: On my side, with 
```typescript 
const bl=await provider.getBlockWithTxHashes();
```

### https://starknet-sepolia.public.blastapi.io/rpc/v0_7
```
{
  l1_da_mode: 'BLOB',
  l1_data_gas_price: { price_in_fri: '0x91acb74c302d3', price_in_wei: '0x14f44f92900' },
  l1_gas_price: { price_in_fri: '0x2d7f150f9892', price_in_wei: '0x68b5b1857' },
  parent_hash: '0x16cc570bf6841d72a650ac06eefe965f8325c01d80df0ed5de2d3b03d62c6ec',
  sequencer_address: '0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8',
  starknet_version: '0.13.1',
  status: 'PENDING',
  timestamp: 1712319031,
  transactions: [
```

### https://starknet-sepolia.public.blastapi.io/rpc/v0_6
```
{
  l1_gas_price: { price_in_fri: '0x2782d833b2c3', price_in_wei: '0x5b07580b4' },
  parent_hash: '0x2ef3f69a56a90ca2864d4679aa64e857b72b7c9702fb3e3caaf07d863da79d1',
  sequencer_address: '0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8',
  starknet_version: '0.13.1',
  status: 'PENDING',
  timestamp: 1712319274,
  transactions: [
```

So, it should work.

Comment (princeibs) [reply src/provider/rpc.ts]: Fixed ✅. What caused the problem earlier was because I was using older RPC version. @PhilippeR26

**Pull request merged**